File: Java/Platform/Basics/SerializationTest/src/Site.java
package tourism;

import java.util.*;

public class Site implements java.io.Serializable {

	String name;
	private List<Visitor> visitors;

	Site(String name) {
		this.name = name;
		visitors = new ArrayList<>();
	}

	public final String getName() { return name; }

	public final List<Visitor> getVisitors() { return visitors; }

	public Visitor getVisitor(String id) {
		return visitors.stream()
				.filter(v -> v.id.equals(id))
				.findFirst()
				.orElseGet(() -> {
					Visitor v = new Visitor(id);
					visitors.add(v);
					return v;
				});
	}

	static final long serialVersionUID = 1L;
}

File: dotnet/Platform/Basics/SerializationTest/Tourism/Site.cs
namespace Tourism;

public class Site
{
    public string? Id { get; set; }

    public List<Visitor> Visitors { get; set; } = new();

    public Visitor GetVisitor(string name)
    {
        Visitor? visitor = Visitors.Find(v => v.Id == name);
        if(visitor is null)
        {
            visitor = new Visitor { Id = name };
            Visitors.Add(visitor);
        }
        return visitor;
    }
}